### ISSUE TITLE
encoding/mvt fails on nil features

### ISSUE THREAD

**Issue by polastre** [paulmach/orb#135]
When a layer has an empty feature, often due to clipping, it causes `MarshalGzipped` (and probably `Marshal` too) to fail with a null pointer panic. The panic is on accessing `f.Geometry` in the marshaller. On further investigation, this happens when the geometry or feature is `nil`.

My workflow looks a bit like this:

```golang
// get tileData from source
layers, err := mvt.UnmarshalGzipped(tileData)
if err != nil {
	return nil, err
}
layers.ProjectToWGS84(tileInfo)
layers.Clip(requestedTile.Bound())
layers.ProjectToTile(requestedTile)
data, err := mvt.MarshalGzipped(layers)
```

`mvt.MarshalGzipped(layers)` fails with a panic.

If I add a `Prune` function, something along the lines of:

```golang
func (l *Layer) Prune() {
	feats := []*geojson.Feature{}
	for _, f := range l.Features {
		if f != nil && f.Geometry != nil {
			feats = append(feats, f)
		}
	}
	l.Features = feats
}
```

then the tiles marshal and render correctly.

Can I suggest or open a PR to provide a `Prune` function for both `mvt.Layer` and `mvt.Layers`?

**Comment by paulmach** [comment]
Can these features just be skipped on encoding?

**Comment by polastre** [comment]
How would you propose skipping them in `MarshalGzipped`, which encodes the entire set?

**Comment by paulmach** [comment]
Nil features are "skipped" during encoding. Features clipped to empty/nil are remove from the features array during clipping https://github.com/paulmach/orb/pull/141

**Comment by paulmach** [comment]
I merged the above change, it is in [v0.11.0](https://github.com/paulmach/orb/releases/tag/v0.11.0)

**Comment by polastre** [comment]
Perfect, that's a great solution. Thanks @paulmach!
